handlers/smshandler/adminapi/adminapi.go

```diff
@@ -0,0 +1,323 @@
+		return err
+	}
+	return ctx.Send(resp, bearerstdapi.HTTPstatusCodeOK)
+}
+
+func user(msg *bearerstdapi.BearerStandardAPIdata, ctx *httprouter.HTTPContext) error {
+	var userID types.HexBytes
+	if err := userID.FromString(ctx.URLParam("userid")); err != nil {
+		return err
+	}
+	user, err := storage.User(userID)
+	if err != nil {
+		return err
+	}
+	resp, err := json.MarshalIndent(user, "", " ")
+	if err != nil {
+		return err
+	}
+	return ctx.Send(resp, bearerstdapi.HTTPstatusCodeOK)
+}
+
+type newUserData struct {
+	Phone string `json:"phone"`
+	Extra string `json:"extra"`
+}
+
+func newUser(msg *bearerstdapi.BearerStandardAPIdata, ctx *httprouter.HTTPContext) error {
+	var userID types.HexBytes
+	if err := userID.FromString(ctx.URLParam("userid")); err != nil {
+		return err
+	}
+	if storage.Exists(userID) {
+		return fmt.Errorf("user already exists")
+	}
+	newUser := newUserData{}
+	if err := json.Unmarshal(msg.Data, &newUser); err != nil {
+		return err
+	}
+	if err := storage.AddUser(userID, nil, newUser.Phone, newUser.Extra); err != nil {
+		return err
+	}
+	return ctx.Send([]byte(respOK), bearerstdapi.HTTPstatusCodeOK)
+}
+
+func addElection(msg *bearerstdapi.BearerStandardAPIdata, ctx *httprouter.HTTPContext) error {
+	var userID, electionID types.HexBytes
+	if err := userID.FromString(ctx.URLParam("userid")); err != nil {
+		return err
+	}
+	if err := electionID.FromString(ctx.URLParam("electionid")); err != nil {
+		return err
+	}
+	user, err := storage.User(userID)
+	if err != nil {
+		return err
+	}
+	election := smshandler.UserElection{
+		ElectionID:        electionID,
+		RemainingAttempts: storage.MaxAttempts(),
+	}
+	user.Elections = append(user.Elections, election)
+	if err := storage.UpdateUser(user); err != nil {
+		return err
+	}
+	return ctx.Send([]byte(respOK), bearerstdapi.HTTPstatusCodeOK)
+}
+
+func addAttempt(msg *bearerstdapi.BearerStandardAPIdata, ctx *httprouter.HTTPContext) error {
+	var userID, election types.HexBytes
+	if err := userID.FromString(ctx.URLParam("userid")); err != nil {
+		return err
+	}
+	if err := election.FromString(ctx.URLParam("election")); err != nil {
+		return err
+	}
+	if err := storage.IncreaseAttempt(userID, election); err != nil {
+		return err
+	}
+	return ctx.Send([]byte(respOK), bearerstdapi.HTTPstatusCodeOK)
+}
+
+func setConsumed(msg *bearerstdapi.BearerStandardAPIdata, ctx *httprouter.HTTPContext) error {
+	var userID, election types.HexBytes
+	if err := userID.FromString(ctx.URLParam("userid")); err != nil {
+		return err
+	}
+	if err := election.FromString(ctx.URLParam("election")); err != nil {
+		return err
+	}
+	consumed := ctx.URLParam("consumed") == "true" || ctx.URLParam("consumed") == "1"
+	user, err := storage.User(userID)
+	if err != nil {
+		return err
+	}
+	i := user.FindElection(election)
+	if i == -1 {
+		return fmt.Errorf("election not found for user %s", user.UserID)
+	}
+	user.Elections[i].Consumed = consumed
+	if err := storage.UpdateUser(user); err != nil {
```

Comment:
If `consumed` is not changed the update is made anyways, better to return an error if trying to set the same value ?